### PR TITLE
Add validations to models

### PR DESCRIPTION
**Pull request by altontonn**
# model validations 💎 
I added validations to the following models 🎯 

1️⃣  User model

- The name must not be blank.

- PostsCounter must be an integer greater than or equal to zero.

2️⃣  Post model

- The title must not be blank.

- The title must not exceed 250 characters.

- CommentsCounter must be an integer greater than or equal to zero.

- LikesCounter must be an integer greater than or equal to zero.

### CODE REVIEW

**Review by Whoistolu — COMMENTED** [review]
Hi @altontonn 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### You have submitted a project, that does not have linters installed. Linters helps us with code standards. Kindly install the appropriate [linters](https://github.com/microverseinc/linters-config) before requesting another review.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/models/user.rb [hunk 1]
```diff
@@ -1,7 +1,10 @@
...
-  has_many :comments, foreign_key: 'author_id'
-  has_many :likes, foreign_key: 'author_id'
+  has_many :posts, foreign_key: :author_id
+  has_many :comments, foreign_key: :author_id
+  has_many :likes, foreign_key: :author_id
+
+  validates :name, presence: true
+  validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
 
   def three_recent_post
```
Comment (Kamba56) [inline app/models/user.rb:9]: - [ ] As demanded by the project requirements, kindly add unit test for the `three_recent_post` method. You are expected to add unit tests for **all of your models' methods and validations.**

File: app/models/post.rb [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
   after_save :update_author_post_counter
 
   def update_author_post_counter
-    author.increment!(:posts_counter)
+    user.increment!(:PostsCounter)
   end
 
+  private
+
   def recent_comment
```
Comment (Kamba56) [inline app/models/post.rb:18]: - [ ] As demanded by the project requirements, kindly add unit test for the `recent_comment` method. You are expected to add unit tests for **all of your models' methods and validations.**

**Review by Kamba56 — CHANGES_REQUESTED** [review]
Hi @altontonn, 

Good job so far!!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [x] Great job adding validations in your models👏👏👏
- [x] Great job adding unit tests for all the validations

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AshakaE — APPROVED** [review]
Hi @altontonn ,  

### Highlights ✨ 

- Thanks for making the requested changes 👍 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Proceed to the next level 👏 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**